tests/alarm_repetition/alarm.sh.in

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+#The health directory to put the alarms
+HEALTHDIR="@configdir_POST@/health.d/"
+
+#output directory
+OUTDIR="workdir/"
+
+#url to do download
+MURL="http://localhost:19999/api/v1/alarms?active"
+
+#error messages
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+NOCOLOR='\033[0m'
+
+MYCDIR="$(pwd)"
+CONFFILE="$MYCDIR/netdata.conf"
+
+change_alarm_file() {
+  if [ -f "$1" ]; then
+    rm "$1"
+  fi
+
+  #copy keeping the permissions
+  cp -a "$2" "$3"
+}
+
+netdata_test_download() {
+  OPT="-e"
+  if [ "$3" == "I" ]; then
+    OPT="-v"
+  fi
+
+  grep "HTTP/1.1 200 OK" "$1" 2>/dev/null 1>/dev/null
+  TEST="$?"
+  if [ "$TEST" -ne "0" ]; then
+    echo -e "${RED} Error to get the alarms"
+    killall netdata
+    exit 1
+  fi
+
+  COUNT=$(grep -w "\"last_repeat\":" "$2" | grep -c "$OPT" "\"0\"")
+  if [ "$COUNT" -eq "0" ]; then
+    echo -e "${RED} Netdata gave an unexpected result when alarm repetition is $4"
+    killall netdata
+    exit 1
+  fi
+
+  echo -e "${GREEN} I got the expected result"
+}
+
+get_the_logs() {
+  curl -v -k --create-dirs -o "$OUTDIR/$1.out" "$MURL" 2> "$OUTDIR/$1.err"
+  netdata_test_download "$OUTDIR/$1.err" "$OUTDIR/$1.out" "$2" "$3"
+}
+
+process_data() {
+  SEC=120
+  netdata -c "$CONFFILE"
+  echo -e "${NOCOLOR}Sleeping during $SEC seconds to create alarm entries"
+  sleep $SEC
+  get_the_logs "$1" "$2" "$3"
+  killall netdata
```

Comment:
This `killall netdata` frightens me a little bit, although I understand that those tests should be run in a development environment and not in production. The same thing happens with `tests/acls/`. Ideally I would prefer killing a PID somehow. e.g.
```
netdata -c "$CONFFILE" &
NETDATAPID=$!
...
kill $NETDATAPID
```

Reply:
I used killall, because it is a command that I am using during various years, almost decades, but I agree with you that it can be a problem, because some distributions are not having it more. My unique concern that I will test right now it is the fact that Netdata forks 2 times after the process starts, so I do not know whether we will have the PID kept.

Reply:
In my example I gave the `-D` argument, that should cover it hopefully.

Reply:
My concern about `killall` is that it may kill other `netdata` instances.

Reply:
I did not see the -D, I put it in the final version of the script, thanks